stat/sampleuv/weighted_test.go

```diff
@@ -266,3 +266,48 @@ func TestWeightedNoResample(t *testing.T) {
 		}
 	}
 }
+
```

Comment:
Please add a test for the cases in the issue
```
var issue1866Tests = []struct {
	weights []float64
	seed    uint64
}{
	{weights: []float64{72000, 166.66666666666666}, seed: 202},
	{weights: []float64{0.6666666666666666, 20412, 0}, seed: 0},
}

// See https://github.com/gonum/gonum/issues/1866
func TestIssue1866(t *testing.T) {
	for i, test := range issue1866Tests {
		w := NewWeighted(test.weights, rand.NewSource(test.seed))
		for j := 0; j < len(test.weights)+1; j++ {
			if panicked(func() { w.Take() }) {
				t.Errorf("unexpected panic for test %d iteration %d", i, j)
			}
		}
	}
}

func panicked(fn func()) (yes bool) {
	defer func() {
		if recover() != nil {
			yes = true
		}
	}()
	fn()
	return false
}
```

Reply:
Fixed, but maybe it would be better to prefer subtests with t.Run?

Reply:
This will do; having the test table in the func body makes it more cluttered and the tests are simple enough not to justify sub-tests.

Please move this addition to above the fuzz test.